contrib/node/src/python/pants/contrib/node/tasks/javascript_style.py

```diff
@@ -59,10 +65,7 @@ def get_lintable_node_targets(self, targets):
             target
             for target in targets
             if isinstance(target, NodeModule)
-            and (
-                target.has_sources(self._JS_SOURCE_EXTENSION)
-                or target.has_sources(self._JSX_SOURCE_EXTENSION)
-            )
+            and target.has_sources(tuple(self.get_options().file_extensions))
```

Comment:
I am sort of abusing knowledge of the internals here by passing a tuple to [Target.has_sources()](https://github.com/pantsbuild/pants/blob/55d2d02b8e30a97c8e62f6ddf9b9f57337647354/src/python/pants/build_graph/target.py#L564), that is documented as accepting a string. Internally it is passed to str.endswith which happily accepts tuples. It may make sense to document that in has_sources(), but I don't know how to do it correctly in a docstring. Of course, I can also replace this call with `any(...)`

Reply:
Hm, I agree that it would be better to use `any()` here. This reads a little weird, as you identify, and might confuse future developers.

Reply:
Changed. Also changed tuple to any() in `get_javascript_sources` for consistency